contracts/internal/EverlongERC4626.sol

```diff
@@ -54,6 +34,97 @@ abstract contract EverlongERC4626 is ERC4626 {
         _decimals = success ? result : _DEFAULT_UNDERLYING_DECIMALS;
     }
 
+    // ╭─────────────────────────────────────────────────────────╮
+    // │ Stateful                                                │
+    // ╰─────────────────────────────────────────────────────────╯
+
+    /// @inheritdoc ERC4626
+    function deposit(
+        uint256 assets,
+        address to
+    ) public override returns (uint256 shares) {
+        IERC20(_asset).approve(hyperdrive, assets);
+        shares = super.deposit(assets, to);
+    }
+
+    /// @inheritdoc ERC4626
+    function redeem(
+        uint256 shares,
+        address to,
+        address owner
+    ) public override returns (uint256 assets) {
+        assets = super.redeem(shares, to, owner);
+        rebalance();
+    }
+
+    // TODO: Implement.
+    /// @inheritdoc ERC4626
+    function mint(uint256, address) public override returns (uint256 assets) {
+        revert("mint not implemented, please use deposit");
+    }
+
+    // TODO: Implement.
+    /// @inheritdoc ERC4626
+    function withdraw(
+        uint256,
+        address,
+        address
+    ) public override returns (uint256 shares) {
+        revert("withdraw not implemented, please use redeem");
+    }
+
+    // ╭─────────────────────────────────────────────────────────╮
+    // │ Overrides                                               │
+    // ╰─────────────────────────────────────────────────────────╯
+
+    // TODO: Actually estimate this based on current position contents.
+    //       Currently, this is obtained from tracking deposits - withdrawals
+    //       which results in Everlong taking all profit and users receiving
+    //       exactly what they put in.
+    /// @notice Returns the total value of assets managed by Everlong.
+    /// @return Total managed asset value.
+    function totalAssets() public view override returns (uint256) {
+        return _virtualAssets;
+    }
+
+    /// @inheritdoc ERC4626
+    function maxDeposit(
+        address
+    ) public view override returns (uint256 maxAssets) {
+        maxAssets = HyperdriveUtils.calculateMaxLong(IHyperdrive(hyperdrive));
+    }
+
+    /// @dev Decrement the virtual assets and close sufficient positions to
+    ///      service the withdrawal.
+    /// @param _assets Amount of assets owed to the withdrawer.
+    function _beforeWithdraw(
+        uint256 _assets,
+        uint256
+    ) internal virtual override {
+        _virtualAssets -= _assets;
```

Comment:
This should be commented, and we should add a newline. Based on the style guide, there shouldn't be lines of code like this immediately above a comment block.